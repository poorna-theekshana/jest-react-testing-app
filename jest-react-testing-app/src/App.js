import React, { useState } from "react";

const STATUS = {
  HOVERED: "hovered",
  NORMAL: "normal",
};

const fetchData = () => {
  // Simulate fetching data, for testing purposes
  return Promise.resolve({ data: "mocked data" });
};

export default function App() {
  const [status, setStatus] = useState(STATUS.NORMAL);

  const onMouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setStatus(STATUS.NORMAL);
  };

  return (
    <div
      className={status}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      Hover over me
    </div>
  );
}

export { fetchData };
