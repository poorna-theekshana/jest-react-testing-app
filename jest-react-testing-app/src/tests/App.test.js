import renderer from "react-test-renderer";
import App from "../App";
import React from "react";
import { fetchData } from "../App";
import axios from "axios"; 

it("changes the class when hovered", () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  renderer.act(() => {
    tree.props.onMouseEnter();
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  renderer.act(() => {
    tree.props.onMouseLeave();
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

jest.mock("axios");
jest.useFakeTimers();

test("fetchData resolves correctly", async () => {
  axios.get.mockResolvedValue({ data: "mocked data" });

  const promise = fetchData();
  jest.runAllTimers();

  const result = await promise;
  expect(result.data).toBe("mocked data");
});
