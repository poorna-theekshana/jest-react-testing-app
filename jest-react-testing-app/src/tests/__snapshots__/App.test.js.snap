// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Hover over me
</div>
`;

exports[`changes the class when hovered 2`] = `
<div
  className="hovered"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Hover over me
</div>
`;

exports[`changes the class when hovered 3`] = `
<div
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Hover over me
</div>
`;
